src/apps/service/apps/de/metadata.clj

```diff
@@ -153,19 +155,34 @@
             :value (config/workspace-metadata-beta-attr-label)
             :unit  "attr"}]})
 
+(defn- community-name->admin-set
+  [username community-name]
+  (->> (communities/get-community-admin-set username community-name)
+       (map #(vector community-name %))))
+
+(defn- notify-community-admins
+  [username app-name community-names]
+  (let [admin->communities-map (->> community-names
+                                    (mapcat (partial community-name->admin-set username))
+                                    (group-by second))]
+    (doseq [[admin community+admin-lists] admin->communities-map]
+      (notifications/send-community-admin-notification app-name admin (map first community+admin-lists)))))
```

Comment:
These functions need to transform a list of community names, each with a list of their respective admins, into a list of those unique admins, each with their communities. Then the admins can be notified that the app was added to the communities they administrate.

So given the list of community names extracted from the app's published metadata, first the admins of each community is fetched from the `iplant-groups` service.

For example, given communities `com1`, `com2`, and `com3`, the admins may be `user1` and `user2` for `com1`, `user2` and `user3` for `com2`, and `user3` and `user4` for `com3`. As the admins are fetched for each community, they're combined into a list of community-admin lists, such as `[["com1" "user1"], ["com1" "user2"]]`, then combined into 1 list by `mapcat`:
```clojure
[["com1" "user1"], ["com1" "user2"], ["com2" "user2"], ["com2" "user3"], ["com3" "user3"], ["com3" "user4"]]
```

Then `group-by` turns this into a map keyed by each unique admin:
```clojure
{"user1" [["com1" "user1"]], "user2" [["com1" "user2"] ["com2" "user2"]], "user3" [["com2" "user3"] ["com3" "user3"]], "user4" [["com3" "user4"]]}
```

Finally, `doseq` iterates over each admin-list/key-value pair, passing the admin and the community list (extracted with `map first`) to the notifications function.

I'm open to suggestions if there's a better way to do this 🤔 
Otherwise, I can add some of this explanation to these function docs.